drivers/sensor/lsm6dsl/lsm6dsl.h

```diff
@@ -689,12 +688,23 @@
 #endif
 
 struct lsm6dsl_config {
-	char *i2c_master_dev_name;
-	u16_t i2c_slave_addr;
+	char *comm_master_dev_name;
+};
+
+struct lsm6dsl_data;
+
+struct lsm6dsl_transfer_function {
+	int (*read_data)(struct lsm6dsl_data *data, u8_t reg_addr,
+			u8_t *value, u8_t len);
+	int (*read_reg)(struct lsm6dsl_data *data, u8_t reg_addr,
+			u8_t *value);
+	int (*update_reg)(struct lsm6dsl_data *data, u8_t reg_addr,
+			u8_t value, u8_t mask);
 };
 
 struct lsm6dsl_data {
-	struct device *i2c_master;
+	struct device *comm_master;
+	struct lsm6dsl_config *cfg;
```

Comment:
Is that required to put config into data?
as it is already provided to driver thanks to DEVICE_AND_API_INIT macro

Reply:
It is not. I'll remove it.